mono/metadata/file-io.c

```diff
@@ -489,8 +489,11 @@ ves_icall_System_IO_MonoIO_FindFirst (MonoString *path,
 	ifh = g_new (IncrementalFind, 1);
 	ifh->find_handle = find_handle;
 	ifh->utf8_path = mono_string_to_utf8_checked (path, &error);
-	if (mono_error_set_pending_exception (&error))
+	if (mono_error_set_pending_exception (&error)) {
+		FindClose (find_handle);
```

Comment:
I don't think it would hurt to have them with `MONO_ENTER_GC_SAFE`/`MONO_EXIT_GC_SAFE` around them. So it would simply be like:
```
if (mono_error_set_pending_exception (&error)) {
  MONO_ENTER_GC_SAFE;
  FindClose (find_handle);
  MONO_EXIT_GC_SAFE;
  g_free (ifh);
  return NULL;
}
```